extend ProductProcessor to import product options and variants

olimination opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?   | no
| Fixed tickets | #200

Add support for importing Product variants with their related Product options. 

Current solution notes:

- Product options need to be prefixed in the csv file with "Product_Option_<OPTION_CODE_NAME>"
- The Product options need to exist already in Sylius
- To create different Product variants, you set the same "Code" for each product row, so the relationship between the Product and Product variants can be managed without any additional "Parent Code" column for example.

Still open todo:

- Tests
- Export logic
- Docs update

olimination commented:
Hi @vasilvestre 
not so much anymore I think :)
The core logic is here, we use it already for some customer, till now we haven't discovered any bigger issues, so I assume it works so far :)

What I see what's still missing is:
- Resync with main/master, resolving conflicts, retest
- Update/Write some tests
- Docs
- Export logic -> but here I would rather go for handling this in a separate issue

Unfortunately currently my time is very restricted, not sure when I can finalize the open stuff...sorry for that...

Lovnish commented:
hgjn vmhj

Lovnish commented:
I need to import products with variations. Can you provide me csv file sample with variations

Lovnish commented:
And one more thing these two files in need to replpace in vendor

vasilvestre commented:
You are done with the new commit ? I will ask @oallain for a workflow approval :)

oallain commented:
Workflow is running

olimination commented:
I have rebased the new changes from master to this branch and I have also added a sample csv file in `tests/Behat/Resources/fixtures/products_and_variants.csv` (@Lovnish what you have asked for) and added some doc notes in the README file to explain the behaviour a bit.

I still could not manage to write some automated tests. This is my first time using Behat&Co. (I usually code in Java/Kotlin/Scala :)) so I still need some more time.

@vasilvestre: If you like, you could use the dev dependency and add the repository to your composer.json and use it/test it already for your use case: `"friendsofsylius/sylius-import-export-plugin": "dev-import-export-product-options-and-variants"` till the tests are done. The core logic should be fine so far...but I was just the only tester :)
